src/de.rs

```diff
@@ -5,90 +5,82 @@
 use errors::Error as LibError;
 use errors::ErrorKind;
 use errors::Result as LibResult;
-use neon::js;
-use neon::mem::Handle;
-use neon::scope::Scope;
+use neon::prelude::*;
 use serde;
 use serde::de::Visitor;
-
-use neon::js::Object;
-use neon::js::Value;
-use neon::js::Variant;
-use neon::vm::Lock;
 use serde::de::{DeserializeOwned, DeserializeSeed, EnumAccess, MapAccess, SeqAccess, Unexpected,
                 VariantAccess};
 
-/// Deserialize an instance of type `T` from a `Handle<js::JsValue>`
+/// Deserialize an instance of type `T` from a `Handle<JsValue>`
 ///
 /// # Errors
 ///
 /// Can fail for various reasons see `ErrorKind`
 ///
-pub fn from_value<'j, S, T>(scope: &mut S, value: Handle<'j, js::JsValue>) -> LibResult<T>
+pub fn from_value<'j, T>(cx: &mut CallContext<'j, JsObject>, value: Handle<'j, JsValue>) -> LibResult<T>
 where
-    S: Scope<'j>,
     T: DeserializeOwned + ?Sized,
 {
-    let mut deserializer: Deserializer<S> = Deserializer::new(scope, value);
+    let mut deserializer: Deserializer = Deserializer::new(cx, value);
     let t = T::deserialize(&mut deserializer)?;
     Ok(t)
 }
 
 #[doc(hidden)]
-pub struct Deserializer<'a, 'j, S: Scope<'j> + 'a> {
-    scope: &'a mut S,
-    input: Handle<'j, js::JsValue>,
+pub struct Deserializer<'a, 'j: 'a> {
+    cx: &'a mut FunctionContext<'j>,
```

Comment:
I think these should take generic `Context` instead so that `neon-serde` can be used in other places (e.g., `TaskContext`).